substratevm/src/com.oracle.svm.core.graal.llvm/src/com/oracle/svm/core/graal/llvm/util/LLVMDebugLineInfo.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (c) 2018, 2019, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.svm.core.graal.llvm.util;
+
+import com.oracle.graal.pointsto.infrastructure.OriginalClassProvider;
+import com.oracle.graal.pointsto.infrastructure.WrappedJavaMethod;
+import com.oracle.graal.pointsto.meta.AnalysisType;
+import com.oracle.svm.core.SubstrateOptions;
+import com.oracle.svm.hosted.annotation.CustomSubstitutionMethod;
+import com.oracle.svm.hosted.annotation.CustomSubstitutionType;
+import com.oracle.svm.hosted.image.NativeImage;
+import com.oracle.svm.hosted.image.sources.SourceManager;
+import com.oracle.svm.hosted.lambda.LambdaSubstitutionType;
+import com.oracle.svm.hosted.meta.HostedMethod;
+import com.oracle.svm.hosted.meta.HostedType;
+import com.oracle.svm.hosted.substitute.InjectedFieldsType;
+import com.oracle.svm.hosted.substitute.SubstitutionMethod;
+import com.oracle.svm.hosted.substitute.SubstitutionType;
+import jdk.vm.ci.meta.LineNumberTable;
+import jdk.vm.ci.meta.ResolvedJavaMethod;
+import jdk.vm.ci.meta.ResolvedJavaType;
+import org.graalvm.compiler.debug.DebugContext;
+import org.graalvm.compiler.graph.NodeSourcePosition;
+import org.graalvm.nativeimage.ImageSingletons;
+
+import java.nio.file.Path;
+
+import static org.graalvm.compiler.debug.GraalError.unimplemented;
+
+// Similar to NativeImageDebugLineInfo from previous releases
+public class LLVMDebugLineInfo {
```

Comment:
`NativeImageDebugLineInfo` has been replaced with `NativeImageDebugLocationInfo`. You should base your changes on the newer version to be able to merge the changes to master.

Reply:
Also, is it necessary to reimplement this class? Can't you just reuse it as is?

Reply:
There are a great number of other changes that have happened along with that change.

Reply:
Okay.

I pushed some changes where I attempted to use the standard `NativeImageDebugInfoProvider` backend for the location information. I had to make the class public and add a new subclass `NativeImageDebugLLVMLocationInfo`. 

I had one question regarding `NativeImageBaseMethodInfo` constructor. When I tried using the default constructor, I got a NullPointerException from `isPseudoObjectType()` called from `createParamInfo()` inside the constructor. I am not sure what these functions do, so I don't know how to fix the error for the LLVM backend. I was hoping if someone could let me know what they do. Currently, in the pushed commits, I have a workaround where I created another constructor inside `NativeImageBaseMethodInfo` that does not make these calls.

Let me know if this is similar to what you were expecting.

Reply:
@rishikeshdevsot Method `isPseudoObjectType`  checks whether a type presented as a Java object type is derived from class WordBase. ThaIf so that means the Object type used as a proxy for a foreign numeric or pointer value. As far as the type info model is concerned the debugger still needs to present these types as object types. However, as far as the compiled code is concerned they get handled as 64 bit words. That only has influences debug info generation for local vars and values. A parameter or local var of this type will use the object type as its type but will use long as its JavaKind. A constant value for these types will use a long not an oop. This is important when processing frame data that identifies local values and when checking that repeat occurences of vars and values actually alias each other.

I'm not sure what you mean when you say "When I tried using the default constructor" for `NativeImageBaseMethodInfo`. Do you mean you added a constructor which does not receive a `ResolvedJavaMethod`? If so then the resulting `NullPointerException` is only going to be the start of your problems.

If instead you used the existing constructor and it gave you an NPE then it looks like something is very wrong with the `ResolvedJavaMethod` you passed it.